tests/compile-test.rs

```diff
@@ -45,48 +46,49 @@ fn third_party_crates() -> String {
         "syn",
     ];
     let dep_dir = cargo::TARGET_LIB.join("deps");
-    let mut crates: HashMap<&str, PathBuf> = HashMap::with_capacity(CRATES.len());
-    for entry in fs::read_dir(dep_dir).unwrap() {
-        let path = match entry {
-            Ok(entry) => entry.path(),
-            Err(_) => continue,
-        };
-        if let Some(name) = path.file_name().and_then(OsStr::to_str) {
-            for dep in CRATES {
-                if name.starts_with(&format!("lib{}-", dep))
-                    && name.rsplit('.').next().map(|ext| ext.eq_ignore_ascii_case("rlib")) == Some(true)
-                {
-                    if let Some(old) = crates.insert(dep, path.clone()) {
-                        // Check which action should be done in order to remove compiled deps.
-                        // If pre-installed version of compiler is used, `cargo clean` will do.
-                        // Otherwise (for bootstrapped compiler), the dependencies directory
-                        // must be removed manually.
-                        let suggested_action = if std::env::var_os("RUSTC_BOOTSTRAP").is_some() {
-                            "remove the stageN-tools directory"
-                        } else {
-                            "run `cargo clean`"
-                        };
-
-                        panic!(
-                            "\n---------------------------------------------------\n\n \
-                            Found multiple rlibs for crate `{}`: `{:?}` and `{:?}`.\n \
-                            Probably, you need to {} before running tests again.\n \
-                            \nFor details on that error see https://github.com/rust-lang/rust-clippy/issues/7343 \
-                            \n---------------------------------------------------\n",
-                            dep, old, path, suggested_action
-                        );
-                    }
-                    break;
-                }
-            }
+    let mut crates: HashMap<&str, Vec<PathBuf>> = HashMap::with_capacity(CRATES.len());
+    let mut flags = String::new();
+    for entry in fs::read_dir(dep_dir).unwrap().flatten() {
+        let path = entry.path();
+        if let Some(name) = try {
```

Comment:
Shouldn't this trigger the `clippy::blocks_in_if_condition` lint? :thinking:

Reply:
sounds reasonable